### PR TITLE
[r2r] Add Postman collection for all API methods

### PR DESCRIPTION
**Pull request by shamardy**
#362 

A few notes:

- To start using the postman collection, import all the environments in addition to importing the collection, setup `userpass` for `mm2_local_node_1`, `mm2_local_node_2`, then choose which environment to use in postman.
- I only added a collection for the current AtomicDEX-API dev branch, when merging the dev branch to mm2.1 branch for a new release we should create a new collection for mm2.1 branch that should be a copy of the up to date dev branch collection at that time.
- Before or after merging a PR to dev, if this PR made any change to the API methods or added new methods, the PR author should open a PR to developer-docs with the changes added to the postman collection to keep it up to date, I will be adding this step to [AtomicDEX-API Documentation workflow](https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1/docs/GIT_FLOW_AND_WORKING_PROCESS.md#issues-handling-process-developmentqadocumentation-workflow).

### CODE REVIEW

File: postman/collections/mm2_dev.postman_collection.json [hunk 1]
```diff
@@ -0,0 +1,5098 @@
...
+									"raw": "{{address}}",
+									"host": [
+										"{{address}}"
+									]
+								}
+							},
+							"response": []
+						},
+						{
+							"name": "init_withdraw",
```
Comment (sergeyboyko0791) [inline postman/collections/mm2_dev.postman_collection.json:3826]: What if we move `init_withdraw`, `withdraw_status` and `withdraw_user_action` to a separate `withdraw` folder?

File: postman/collections/mm2_dev.postman_collection.json [hunk 2]
```diff
@@ -0,0 +1,5098 @@
...
+								"header": [
+									{
+										"key": "Content-Type",
+										"value": "application/json",
+										"type": "text"
+									}
+								],
+								"body": {
+									"mode": "raw",
+									"raw": "{\r\n    \"userpass\": \"{{userpass}}\",\r\n    \"mmrpc\": \"2.0\",\r\n    \"method\": \"my_tx_history\",\r\n    \"params\": {\r\n        \"coin\": \"RICK\"\r\n        // \"limit\": 10,\r\n        // \"paging_options\": {\r\n        //     // \"FromId\": null, // Accepted values: Strings\r\n        //     \"PageNumber\": 1 // used only if: \"from_id\": null\r\n        // }\r\n    }\r\n    // \"id\": null // Accepted values: Integers\r\n}"
```
Comment (sergeyboyko0791) [inline postman/collections/mm2_dev.postman_collection.json:3892]: `my_tx_history` is currently supported for BCH/SLP only

File: postman/collections/mm2_dev.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,5098 @@
...
+										"header": [
+											{
+												"key": "Content-Type",
+												"value": "application/json",
+												"type": "text"
+											}
+										],
+										"body": {
+											"mode": "raw",
+											"raw": "{\r\n    \"userpass\": \"{{userpass}}\",\r\n    \"method\": \"electrum\",\r\n    \"coin\": \"ETH\",\r\n    \"urls\": [\r\n        \"http://eth1.cipig.net:8555\",\r\n        \"http://eth2.cipig.net:8555\",\r\n        \"http://eth3.cipig.net:8555\"\r\n    ],\r\n    \"swap_contract_address\": \"0x24ABE4c71FC658C91313b6552cd40cD808b3Ea80\"\r\n    // \"fallback_swap_contract\": null, // Accepted values: Strings\r\n    // \"gas_station_url\": null, // Accepted values: Strings\r\n    // \"gas_station_decimals\": 8,\r\n    // \"gas_station_policy\": {\r\n    //     \"policy\": \"MeanAverageFast\" // Accepted values: \"MeanAverageFast\", \"Average\"\r\n    // },\r\n    // \"mm2\": null, // Required only if: Not in Coin Config // Accepted values: 0, 1\r\n    // \"tx_history\": false,\r\n    // \"required_confirmations\": 1 // Default: Coin Config\r\n}"
```
[the remark this answers is not in the document]
Comment (shamardy) [reply postman/collections/mm2_dev.postman_collection.json:85]: I will remove ETH from `electrum` and just leave it in `enable`

File: postman/collections/mm2_dev.postman_collection.json [hunk 4]
```diff
@@ -0,0 +1,5098 @@
...
+									"raw": "{{address}}",
+									"host": [
+										"{{address}}"
+									]
+								}
+							},
+							"response": []
+						},
+						{
+							"name": "get_trade_fee",
```
[the remark this answers is not in the document]
Comment (shamardy) [reply postman/collections/mm2_dev.postman_collection.json:1806]: I believe so, do you think this and `trade_preimage` should be somewhere else?

File: postman/collections/mm2_dev.postman_collection.json [hunk 5]
```diff
@@ -0,0 +1,5098 @@
...
+												"{{address}}"
+											]
+										}
+									},
+									"response": []
+								}
+							]
+						},
+						{
+							"name": "Hardware Wallet",
```
[the remark this answers is not in the document]
Comment (shamardy) [reply postman/collections/mm2_dev.postman_collection.json:3470]: I think the purpose of this collection is to keep it inline with the dev branch at all times, that's why I covered all the methods in dev. When the HD API is finished or updated with new PRs merging to dev we should remove the removed methods and update the collection accordingly.

[the remark this answers is not in the document]
Comment (sergeyboyko0791) [reply postman/collections/mm2_dev.postman_collection.json:3431]: Ooh I see. I hope to finish the refactoring before the Postman collection goes public.

File: postman/collections/mm2_dev.postman_collection.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergeyboyko0791) [reply postman/collections/mm2_dev.postman_collection.json:1767]: What do you think if we move them to `Orders` folder? 🤔 
They are involved to be used before `buy`/`sell`/`setprice`

[the remark this answers is not in the document]
Comment (shamardy) [reply postman/collections/mm2_dev.postman_collection.json:1767]: It makes sense as `trade_preimage` has a `swap_method` field that's one of `setprice`, `buy`, `sell`

**Comment by smk762** [comment]
Nice work :) Set it up without any problems.
Only thing I've noticed worth noting so far is some newer methods not working on playground node (e.g. sign_message)
`{"mmrpc":"2.0","error":"No such method","error_path":"dispatcher","error_trace":"dispatcher:182]","error_type":"NoSuchMethod","id":null}` though I assume these will filter thru eventually.

**Comment by shamardy** [comment]
> Only thing I've noticed worth noting so far is some newer methods not working on playground node (e.g. sign_message)

When requesting mm2 version for the playground node using the `version` method it returns
`
{
    "result": "2.1.6168_continuous-container-push_87b51bc1a_Linux_Release",
    "datetime": "2022-05-05T19:15:01+03:00"
}
`
The sign message PR was commited to dev after this date https://github.com/KomodoPlatform/atomicDEX-API/commit/128d0965335449bad8a22aaed9a99a5f92110b5a

@ozkanonur isn't the playground node supposed to be automatically deployed? am I using the correct ip address and ports in the postman environments?

**Comment by ozkanonur** [comment]
> > Only thing I've noticed worth noting so far is some newer methods not working on playground node (e.g. sign_message)
> 
> When requesting mm2 version for the playground node using the `version` method it returns `{ "result": "2.1.6168_continuous-container-push_87b51bc1a_Linux_Release", "datetime": "2022-05-05T19:15:01+03:00" }` The sign message PR was merged to dev after this date [KomodoPlatform/atomicDEX-API@128d096](https://github.com/KomodoPlatform/atomicDEX-API/commit/128d0965335449bad8a22aaed9a99a5f92110b5a)
> 
> @ozkanonur isn't the playground node supposed to be automatically deployed? am I using the correct ip address and ports in the postman environments?

It does redeploy all the time, but I forgot to set image pull policy. Nodes wasn't pulling the latest version of the image because they already exist in the machine. I updated the deployment file, should be fixed now.

**Comment by shamardy** [comment]
> It does redeploy all the time, but I forgot to set image pull policy. Nodes wasn't pulling the latest version of the image because they already exist in the machine. I updated the deployment file, should be fixed now

Works fine now, this is the response for the `version` method request
`{
    "result": "2.1.6685_dev_312cf45a0_Linux_Release",
    "datetime": "2022-06-28T17:53:34+07:00"
}`

**Comment by smk762** [comment]
Does the `coins` file also update periodically on the playground nodes?
`sign_message` method is recognised by mm2 now (thanks for updating!) but I have a new error - 
`{"mmrpc":"2.0","error":"sign_message_prefix is not set in coin config","error_path":"utxo_common","error_trace":"utxo_common:1707]","error_type":"PrefixNotFound","id":null}`

The PR to add these to coins file has not yet been merged - https://github.com/KomodoPlatform/coins/pull/463 so if auto update is implemented it should work once that is in master branch of coins repo

**Comment by ozkanonur** [comment]
> The PR to add these to

No, there is no relation between coins repository and deployments. Coins configurations are in the deployment repository. Each deployment has it's own coins configuration file.

**Review by artemii235 — APPROVED** [review]
Disregard this comment.

**Review by artemii235 — COMMENTED** [review]
It seems that there is only the collection itself. I would like to see it also as a snippet on the website 🙂

**Comment by gcharang** [comment]
@artemii235 if the data in these files is good, let's merge this PR. I will make the data in these files displayed on the website in a different PR

**Comment by shamardy** [comment]
> It seems that there is only the collection itself. I would like to see it also as a snippet on the website 🙂

Can you please clarify what do you mean by snippet? There are already snippets as part of the APIs documentations.

@gcharang @artemii235 My thinking for the playground nodes and this collection is to one day do requests through the website through a new playground interface with selections of which optional parameters to include and dropdowns for different accepted values etc.. as written in my last point in this comment here https://github.com/KomodoPlatform/developer-docs/issues/362#issuecomment-1101299392

**Comment by gcharang** [comment]
>  to one day do requests through the website through a new playground interface with selections of which optional parameters to include and dropdowns for different accepted values etc.. 

@shamardy till then, people can import the postman collection and send requests to the playground nodes described in the env files right? if so, we can write a doc explaining how to use the collection and how to send requests to the playground nodes

**Comment by shamardy** [comment]
> @shamardy till then, people can import the postman collection and send requests to the playground nodes described in the env files right?

That's correct

> if so, we can write a doc explaining how to use the collection and how to send requests to the playground nodes

That's a good idea 🙂

File: postman/collections/mm2_dev.postman_collection.json [hunk 6]
```diff
@@ -0,0 +1,5064 @@
...
+											"raw": "{{address}}",
+											"host": [
+												"{{address}}"
+											]
+										}
+									},
+									"response": []
+								},
+								{
+									"name": "electrum BCH",
```
Comment (artemii235) [inline postman/collections/mm2_dev.postman_collection.json:136]: I prefer to remove legacy BCH and SLP activation or at least mark related methods as deprecated to encourage using `enable_bch_with_tokens` and `enable_slp` v2 methods.

**Review by artemii235 — CHANGES_REQUESTED** [review]
One note.

File: postman/collections/mm2_dev.postman_collection.json (hunk 6 above)
Comment (shamardy) [reply postman/collections/mm2_dev.postman_collection.json]: Done

**Review by artemii235 — APPROVED** [review]
🔥

**Review by gcharang — APPROVED** [review]

**Pull request merged**